### PR TITLE
Ecosystem section

### PR DESCRIPTION
**Pull request by oraNod**
This PR addresses issue #79 by adding an ecosystem section to the homepage with content from the wireframes.

Note that this PR introduces some reusable components along with the ecosystem section. I propose that we focus on getting content complete and worry about styling in separate issues.

### CODE REVIEW

**Comment by samccann** [comment]
This is looking good! Built locally it looks like this (on firefox)
image

So the white learn more buttons bleed into the general white background. And there's an odd 'back to the top' at the bottom of the section.

**Comment by oraNod** [comment]
Thanks @samccann I'll add a border to the learn more button but would prefer not to spend too much time polishing individual styles in the PRs.

Also the back to top thing is a reusable part that we can include in the other sections. You can see it on the wireframe:

image

It links back to the "top" anchor that I added in this PR. I know it looks kind of janky right now but, again, that's something I think we can style separately.

**Comment by samccann** [comment]
yeah it's more a style nit on the back to the top. Needs a bit more spacing, but will approve the PR so it's not holding things up.

**Review by samccann — DISMISSED** [review]
Can't speak to the code parts but the rest LGTM

**Comment by oraNod** [comment]
@samccann Hey, thanks for the approval but I just pushed another change that means you'll need to approve again. I tried to fix those couple nits you had and also rebased on the other PR with the svg icons.

**Review by samccann — APPROVED** [review]
Looking better, thanks!

**Review by GregSutcliffe — DISMISSED** [review]
Looking good, progress! Builds fine on both Firefox and Chromium, and shrinks well if I make a narrow window (unlike the other two bands :stuck_out_tongue:) so good work there.

As noted by @samccann, the border around the boxes needs more padding, and perhaps a hrule between sections? It seems very close to the Subscribe band... Happy to address this in a future PR though

File: templates/homepage-back-to-top.tmpl
[the remark this answers is not in the document]
Comment (oraNod) [reply templates/homepage-back-to-top.tmpl:1]: @GregSutcliffe how strongly do you feel on that one? that's really much of a muchness imo

[the remark this answers is not in the document]
Comment (oraNod) [reply templates/homepage-back-to-top.tmpl:1]: on 2nd thought I'd prefer to keep `homepage-` in the file name so it's easier to match with the corresponding `sass/homepage-` files.

[the remark this answers is not in the document]
Comment (GregSutcliffe) [reply templates/homepage-back-to-top.tmpl:1]: I'm definitely on the fence, just figured I should ask. No worries.

File: templates/homepage-ecosystem-band.tmpl [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<div class="grid-wrapper homepage-ecosystem-band">
+  <div class="width-12-12 width-12-12-m">
+    <div class="section-title">
+      <img class="section-icon" src="/assets/images/communities.svg" />
+      <h2>{{ homepage.ecosystem.title }}</h2>
+      <p>{{ homepage.ecosystem.intro }}</p>
+    </div>
```
[the remark this answers is not in the document]
Comment (GregSutcliffe) [reply templates/homepage-ecosystem-band.tmpl:7]: That seems reasonable, yeah

[the remark this answers is not in the document]
Comment (oraNod) [reply templates/homepage-ecosystem-band.tmpl:12]: @GregSutcliffe added a view more link. also spotted some weirdness in the template that I didn't catch before so it was good to sort that

**Review by GregSutcliffe — APPROVED** [review]

**Comment by GregSutcliffe** [comment]
Thanks @oraNod, this looks good. As noted, there's a general CSS/styling pass to be done, but the content is in the right shape. Thanks!

**Pull request merged**